src/tools/run-make-support/src/rustc.rs

```diff
@@ -4,12 +4,21 @@ use std::path::Path;
 
 use crate::{command, cwd, env_var, set_host_rpath};
 
-/// Construct a new `rustc` invocation.
+/// Construct a new `rustc` invocation. This will automatically set the library
+/// search path as `-L cwd()`. Use [`bare_rustc`] to avoid this.
 #[track_caller]
 pub fn rustc() -> Rustc {
     Rustc::new()
 }
 
+/// Construct a plain `rustc` invocation with no flags set. Note that compiletest
+/// still presets the environment variables `HOST_RPATH_DIR` and `TARGET_RPATH_DIR`
+/// by default.
```

Comment:
Suggestion: sorry I should've been more clear. What I meant was moreso `set_common` calls `set_host_rpath`, not the one in compiletest:

https://github.com/rust-lang/rust/blob/4a31a6c32a9a0e726516b1086f33e3d4a37dc94c/src/tools/run-make-support/src/lib.rs#L363-L375